src/main/java/de/tum/in/www1/artemis/service/UserService.java

```diff
@@ -347,33 +369,52 @@ public void updateUser(String firstName, String lastName, String email, String l
     /**
      * Update all information for a specific user, and return the modified user.
      *
-     * @param userDTO user to update
+     * @param user The user that should get updated
+     * @param updatedUserDTO The DTO containing the to be updated values
      * @return updated user
      */
-    public Optional<UserDTO> updateUser(UserDTO userDTO) {
-        return Optional.of(userRepository.findById(userDTO.getId())).filter(Optional::isPresent).map(Optional::get).map(user -> {
-            this.clearUserCaches(user);
-            user.setLogin(userDTO.getLogin().toLowerCase());
-            user.setFirstName(userDTO.getFirstName());
-            user.setLastName(userDTO.getLastName());
-            user.setEmail(userDTO.getEmail().toLowerCase());
-            user.setImageUrl(userDTO.getImageUrl());
-            user.setActivated(userDTO.isActivated());
-            user.setLangKey(userDTO.getLangKey());
-            Set<Authority> managedAuthorities = user.getAuthorities();
-            managedAuthorities.clear();
-            userDTO.getAuthorities().stream().map(authorityRepository::findById).filter(Optional::isPresent).map(Optional::get).forEach(managedAuthorities::add);
-            this.clearUserCaches(user);
-            log.debug("Changed Information for User: {}", user);
-            return user;
-        }).map(UserDTO::new);
+    public User updateUser(User user, ManagedUserVM updatedUserDTO) {
+        final var oldGroups = user.getGroups();
+        this.clearUserCaches(user);
+        user.setLogin(updatedUserDTO.getLogin().toLowerCase());
+        user.setFirstName(updatedUserDTO.getFirstName());
+        user.setLastName(updatedUserDTO.getLastName());
+        user.setEmail(updatedUserDTO.getEmail().toLowerCase());
+        user.setImageUrl(updatedUserDTO.getImageUrl());
+        user.setActivated(updatedUserDTO.isActivated());
+        user.setLangKey(updatedUserDTO.getLangKey());
+        user.setGroups(updatedUserDTO.getGroups());
+        if (updatedUserDTO.getPassword() != null) {
+            user.setPassword(passwordEncoder().encode(updatedUserDTO.getPassword()));
+        }
+        Set<Authority> managedAuthorities = user.getAuthorities();
+        managedAuthorities.clear();
+        updatedUserDTO.getAuthorities().stream().map(authorityRepository::findById).filter(Optional::isPresent).map(Optional::get).forEach(managedAuthorities::add);
+        user = userRepository.save(user);
+        this.clearUserCaches(user);
+
+        forwardUserUpdates(user, oldGroups);
+
+        log.debug("Changed Information for User: {}", user);
+        return user;
+    }
+
+    private void forwardUserUpdates(User user, Set<String> oldGroups) {
```

Comment:
this name sounds strange. What does it mean? please find a more descriptive one
I guess JavaDoc would be good here.

Reply:
I changed the name and added some JavaDoc